commands/init.go

```diff
@@ -44,26 +58,58 @@ func withClient(fn func(c client.Client, cmd *cobra.Command, args []string) erro
 	}
 }
 
-func InitClientWithUsernameAndPassword(username, password, instanceURL string) (*model.Client4, string, error) {
+func NewAPIv4Client(instanceURL string, allowInsecure bool) *model.Client4 {
 	client := model.NewAPIv4Client(instanceURL)
+
+	if !allowInsecure {
+		transport := &http.Transport{
+			TLSClientConfig: &tls.Config{
+				VerifyPeerCertificate: func(rawCerts [][]byte, verifiedChains [][]*x509.Certificate) error {
+					// loop over certificate chains
+					for _, chain := range verifiedChains {
+						// check all certs but the root one
+						certs := chain[:len(chain)-1]
+
+						for _, cert := range certs {
+							for _, alg := range insecureSignatureAlgorithms {
+								if cert.SignatureAlgorithm == alg {
+									return fmt.Errorf("insecure algorithm %s found in the certificate chain. Use --insecure-sha1-intermediate flag to ignore. Aborting", alg)
```

Comment:
In case there are multiple chains, we only need one to be valid. This will fail if even a single one is invalid.

Reply:
Updated the logic and added tests, PTAL @jupenur